contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_be_valid_tcp_port.py

```diff
@@ -0,0 +1,137 @@
+from socket import getservbyport
+from typing import Optional
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.execution_engine import PandasExecutionEngine
+from great_expectations.expectations.expectation import ColumnMapExpectation
+from great_expectations.expectations.metrics import (
+    ColumnMapMetricProvider,
+    column_condition_partial,
+)
+
+
+# This method compares a string to the valid TCP port number
+def is_valid_tcp_port(port: str) -> bool:
+    try:
+        getservbyport(int(port), "tcp")
+        return True
+    except (ValueError, OSError, OverflowError):
+        return False
+
+
+# This class defines a Metric to support your Expectation.
+# For most ColumnMapExpectations, the main business logic for calculation will live in this class.
+class ColumnValuesToBeValidTcpPort(ColumnMapMetricProvider):
+
+    # This is the id string that will be used to reference your metric.
+    condition_metric_name = "column_values.valid_tcp_port"
+
+    # This method implements the core logic for the PandasExecutionEngine
+    @column_condition_partial(engine=PandasExecutionEngine)
+    def _pandas(cls, column, **kwargs):
+        return column.apply(lambda x: is_valid_tcp_port(x))
+
+    # This method defines the business logic for evaluating your metric when using a SqlAlchemyExecutionEngine
+    # @column_condition_partial(engine=SqlAlchemyExecutionEngine)
+    # def _sqlalchemy(cls, column, _dialect, **kwargs):
+    #     raise NotImplementedError
+
+    # This method defines the business logic for evaluating your metric when using a SparkDFExecutionEngine
+    # @column_condition_partial(engine=SparkDFExecutionEngine)
+    # def _spark(cls, column, **kwargs):
+    #     raise NotImplementedError
+
+
+# This class defines the Expectation itself
+class ExpectColumnValuesToBeValidTcpPort(ColumnMapExpectation):
+    """This Expectation validates data as conforming to the valid TCP port number."""
+
+    # These examples will be shown in the public gallery.
+    # They will also be executed as unit tests for your Expectation.
+    examples = [
+        {
+            "data": {
+                "well_formed_tcp_port": [
+                    "7",  # Echo Protocol
+                    "20",  # File Transfer Protocol (FTP)
+                    "22",  # Secure Shell (SSH)
+                    "179",  # Border Gateway Protocol (BGP)
+                    "853",  # DNS over TLS

```

Comment:
`basic_positive_test` is failing on this value

Reply:
Hi @austiezr Thanks for pointing out! I think the problem is a little more tricky. Just to make sure, what is your testing environment and which value fails? Is it port 853 (DNS over TLS)?

I developed it under the WSL2 in Windows and the test passed successfully. Python implementation of `getservbyport` in socket is a wrapper of [C function](https://docs.microsoft.com/zh-cn/windows/win32/api/winsock/nf-winsock-getservbyport?redirectedfrom=MSDN), which is OS-dependent. It could be a problem in application.

To quickly fix the problem, I change the test case of port 853 to port 37, which is more generally accepted.

Reply:
Interesting! Thanks for reaching back w/ this context. It was port 853 that was failing for me, and the updated value is passing; appreciate your flexibility here. Will approve and merge!